client/extensions/woocommerce/app/products/product-form-variations-modal.js

```diff
@@ -0,0 +1,132 @@
+import VerticalMenu from 'components/vertical-menu';
+
+class ProductFormVariationsModal extends React.Component {
+
+	static propTypes = {
+		product: PropTypes.object.isRequired,
+		variations: PropTypes.array.isRequired,
+		editProductVariation: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedVariation: this.props.selectedVariation,
+		};
+
+		this.switchVariation = this.switchVariation.bind( this );
+		this.setDescription = this.setDescription.bind( this );
+		this.toggleVisible = this.toggleVisible.bind( this );
+	}
+
+	setDescription( e ) {
+		const { selectedVariation } = this.state;
+		const { product, editProductVariation, variations } = this.props;
+		const variation = find( variations, ( v ) => selectedVariation === v.id );
+		editProductVariation( product, variation, { description: e.target.value } );
+	}
+
+	toggleVisible() {
+		const { selectedVariation } = this.state;
+		const { product, editProductVariation, variations } = this.props;
+		const variation = find( variations, ( v ) => selectedVariation === v.id );
+		editProductVariation( product, variation, { visible: ! variation.visible } );
+	}
+
+	switchVariation( selectedVariation ) {
+		this.setState( {
+			selectedVariation
+		} );
+	}
+
+	render() {
+		const { variations, translate } = this.props;
+		const { selectedVariation } = this.state;
+		const variation = find( variations, ( v ) => selectedVariation === v.id );
+
+		const navVariations = ( variations && variations.filter( v => v.attributes.length > 0 ) ) || [];
+		const navItems = navVariations.map( function( v, i ) {
+			return (
+				<ModalNavItem key={ i } variation={ v } selected={ selectedVariation } />
+			);
+		} );
+
+		return (
+			<div className="products__product-form-modal-wrapper">
+				<h1>{ translate( 'Variation details' ) }</h1>
+
+				<VerticalMenu onClick={ this.switchVariation } className="products__product-form-modal-menu">
+					{navItems}
+				</VerticalMenu>
+
+				<div className="products__product-form-modal-contents">
+					<h2>{ formattedVariationName( variation ) }</h2>
+					<span className="products__product-form-sku">SKU:</span>
+
+					<FormFieldSet className="products__product-form-variation-description">
+						<FormLabel htmlFor="description">{ translate( 'Description' ) }</FormLabel>
+						<FormTextArea name="description" value={ variation.description || '' } onChange={ this.setDescription } />
+						<FormSettingExplanation>{ translate(
+								'This will be displayed in addition to the main product description when this variation is selected.'
+						) }</FormSettingExplanation>
+					</FormFieldSet>
+
+					<FormLabel>
+						{ translate( 'Visible' ) }
+						<FormToggle
+							onChange={ this.toggleVisible }
+							checked={ variation.visible }
+						/>
+					</FormLabel>
+					<FormSettingExplanation>{ translate(
+						'Hidden variations cannot be selected for purchase by customers.'
```

Comment:
I wonder if we should only display this warning-like text if they toggle visible to false? cc @kellychoffman - when I see it right now I take an extra moment to think about it and make sure the toggle seems to be enabled